### ISSUE TITLE
Implement PSP8 Pseudopotential format

### ISSUE THREAD

**Issue by mfherbst** [mfherbst/DFTK.jl#2]
The [PSP8 pseudopotential format](https://docs.abinit.org/developers/psp8_info/#information-on-the-format-8-for-pseudopotentials) seems to be worth supporting for experimentation.

**Comment by antoine-levitt** [comment]
There are a few formats around. Any format for which we can get good pseudos (ONCVPSP for physical accuracy, and Goedecker's for mathematical niceness) is good, so it comes down to difficulty of writing a parser. Also it's probable somebody wrote a parser in a sane language (eg python) from which we can maybe steal.

**Comment by antoine-levitt** [comment]
Just looked into this, it doesn't look *that* bad. Additional complications (on top of the usual "figuring out what they wrote in the 'spec'") wrt what we have is a possible core correction (to be checked, but I think it's just a radial density that needs to be added to the pseudo-density in the exchange-correlation term), and numerical potentials/projectors for which we'll need to do quadrature.

**Comment by mfherbst** [comment]
I agree. I looked at them a while back, but then what made me drop it was the bad spec. ONCVPSP are better specified ...

**Comment by azadoks** [comment]
Should follow easily from work done in #741 . Once NLCCs are implemented, the only other feature of PSP8 that's missing is spin-orbit coupling.

**Comment by antoine-levitt** [comment]
Do we need psp8 support? Can't we use external converters from psp8 to upf?

**Comment by azadoks** [comment]
I'm not sure if one exists, but I think the most used psp8 pseudos are those from PseudoDojo which are also available as UPFs (not converted, just also output by ONCVPSP afaik).

**Comment by Technici4n** [comment]
Implemented in https://github.com/JuliaMolSim/DFTK.jl/pull/1160 by converting them to upf. 😉
